lib/errors.js

```diff
@@ -1,8 +1,10 @@
 export class ParseError extends Error {
   #wasConstructedWithAst = false
+  #node
   constructor(message, node, cause) {
     super(message, {cause})
     this.name = 'ParseError'
+    this.node = node
```

Comment:
```suggestion
    this.#node = node
```

I'd prefer a getter additional to this to prevent exposing the values in loggers.
```js
get node () {
  return this.#node
}
```